cmd/k8sagent/unit/agent.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2012-2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package unit
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+	"github.com/juju/names/v4"
+	"github.com/juju/utils/voyeur"
+	"github.com/juju/worker/v2"
+	"github.com/juju/worker/v2/dependency"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/uniter"
+	jujucmd "github.com/juju/juju/cmd"
+	jujudagent "github.com/juju/juju/cmd/jujud/agent"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
```

Comment:
We probably want to avoid bringing in cmdutil if we can because of the mongo dep.

Reply:
Yeah, so just implemented the AgentDone handler inside this package to avoid importing from juud package.